subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/BuildScriptCompileAvoidanceIntegrationTest.kt

```diff
@@ -0,0 +1,389 @@
+package org.gradle.kotlin.dsl.integration
+
+import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
+import org.gradle.kotlin.dsl.fixtures.AbstractKotlinIntegrationTest
+import org.gradle.kotlin.dsl.integration.BuildScriptCompileAvoidanceIntegrationTest.CacheBuster.cacheBuster
+import org.hamcrest.CoreMatchers
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+
+class BuildScriptCompileAvoidanceIntegrationTest : AbstractKotlinIntegrationTest() {
+
+    object CacheBuster {
```

Comment:
❓ Maybe the test should run with a custom script cache directory instead?

How about making `ScriptCacheRepository` accept a custom cache root directory via a system property the way `DefaultGeneratedGradleJarCache` does it? (see https://github.com/gradle/gradle/blob/5808250a35435bb95ec0b357374f50de1e65b1e8/subprojects/core/src/main/java/org/gradle/cache/internal/DefaultGeneratedGradleJarCache.java#L35)

The custom root directory could then be passed as a java.io.File scope here, https://github.com/gradle/gradle/blob/ae794708012c14b2dd636573d96bc5ae030e5d9c/subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/provider/ScriptCacheRepository.kt#L29